### PR TITLE
Added commands for owner to manage permission groups

### PR DESCRIPTION
**Pull request by jeffmomo**
Added 3 commands for managing permission group members from chat

Adds users to a permission group
!adduser <group> <userid>...

Removes users from permission group
!removeuser <group> <userid>...

Shows userids for permission group
!showids <group>

### CODE REVIEW

File: musicbot/bot.py [hunk 1]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+        Usage:
+            {command_prefix}adduser <group> <userid>...
+
+        Adds the user ID(s) to the permission group
+        """
+
+        modify_count = 0
+
+        if not group:
+            return Response('The parameter "group" is required', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 2]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+
+        matching_groups = [x for x in self.permissions.groups if x.name == group]
+
+        if len(matching_groups) == 0:
+            return Response('The given group is not defined in permissions.ini', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 3]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+
+        if not user_id:
+            return Response('User ids not provided', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 4]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+
+        try:
+            int(user_id)
+            if matching_groups[0].add_user(user_id):
+                modify_count += 1
+        except ValueError:
+            return Response('User id is not a number', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 5]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+
+        for id in leftover_args:
+            try:
+                int(id)
+            except ValueError:
+                return Response('id is not a number: %s' % id, reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 6]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+    @owner_only
+    async def cmd_showids(self, message, group):
+        """
+        Usage:
+            {command_prefix}showids <group>
+
+        Shows the user ids in a permission group
+        """
+        if not group:
+            return Response('The parameter "group" is required', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 7]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+
+        matching_groups = [x for x in self.permissions.groups if x.name == group]
+
+        if len(matching_groups) == 0:
+            return Response('The given group is not defined in permissions.ini', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 8]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+        """
+        modify_count = 0
+
+        if not group:
+            return Response('The parameter "group" is required', reply=True, delete_after=20)
+
+        matching_groups = [x for x in self.permissions.groups if x.name == group]
+
+        if len(matching_groups) == 0:
+            return Response('The given group is not defined in permissions.ini', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 9]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+
+        if not user_id:
+            return Response('User ids not provided', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 10]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+
+        try:
+            int(user_id)
+            if matching_groups[0].remove_user(user_id):
+                modify_count += 1
+        except ValueError:
+            return Response('User id is not a number', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 11]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+
+        for id in leftover_args:
+            try:
+                int(id)
+            except ValueError:
+                return Response('id is not a number: %s' % id, reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

File: musicbot/bot.py [hunk 12]
```diff
@@ -885,6 +885,114 @@ def run(self):
...
+        """
+        Usage:
+            {command_prefix}removeuser <group> <userids>...
+
+        Remove the user ID(s) from the permission group
+        """
+        modify_count = 0
+
+        if not group:
+            return Response('The parameter "group" is required', reply=True, delete_after=20)
```
Comment (jaydenkieran) [inline musicbot/bot.py]: The command will fail if the parameter isn't provided. The bot already provides this functionality.

File: musicbot/bot.py (hunk 11 above)
Comment (jaydenkieran) [inline musicbot/bot.py]: This should be raised as a CommandError exception

**Comment by jayktaylor** [comment]
I think your methods are coded badly and don't meet a good quality standard.

Additionally, you're responding with error messages when there's no parameter but the bot's code already does this, and you're not correctly responding with error messages. They should be raised as a CommandError exception.

**Comment by Injusticecraft** [comment]
Has the Conflicting file been fixed yet?

**Comment by jayktaylor** [comment]
As it has been 2 months without activity on this PR, it will be closed. If you wish to PR this feature again, please merge your changes and PR to the `review` branch.

**Pull request closed without merging**